### ISSUE TITLE
MT3620_RTApp_BareMetal_HelloWorld does not build

### ISSUE THREAD

**Issue by matsujirushi** [MediaTek-Labs/mt3620_m4_software#32]
When create CMake cache, I got error below.
```
[CMake] CMake Error: The source directory
```

![image](https://user-images.githubusercontent.com/8079404/91650763-6fe22980-eabe-11ea-865d-4fa2b3a8db29.png)

[log.txt](https://github.com/MediaTek-Labs/mt3620_m4_software/files/5146309/log.txt)

Azure Sphere SDK is 20.07.

**Comment by kevinwh-chou** [comment]
Hi, 

The project on the github should be ok, but similar error message seems to pop up in some building environments in some other case.

Can you test the following step first to see if it works in your environment or not?
Replace the original CMakeLists.txt and CMakeSettings.json in the app folder with the attached ones, then generate cmake cache.
[cmake.zip](https://github.com/MediaTek-Labs/mt3620_m4_software/files/5149378/cmake.zip)

If the error still exists, remove the following lines in the updated CMakeSettings.json and generate cache again.
```
        {
          "name": "AZURE_SPHERE_DEFAULT_CMAKE_PATH",
          "value": "${env.AzureSphereDefaultSDKDir}CMakeFiles\\"
        }
```
Let me know whether it helps.

**Comment by matsujirushi** [comment]
Hi,
Thank you for reply.

> Replace the original CMakeLists.txt and CMakeSettings.json in the app folder with the attached ones, then generate cmake cache.

![image](https://user-images.githubusercontent.com/8079404/91715719-32b68e00-ebc9-11ea-8c68-40ed06243a56.png)

> If the error still exists, remove the following lines in the updated CMakeSettings.json and generate cache again.

![image](https://user-images.githubusercontent.com/8079404/91715740-43670400-ebc9-11ea-82c0-e56b781af1d6.png)


[error2.txt](https://github.com/MediaTek-Labs/mt3620_m4_software/files/5150044/error2.txt)

**Comment by kevinwh-chou** [comment]
Hi,

Thanks for the update, then it looks to be a different case. 
This kind of error seems to occur in some build environments after upgrading Azure Sphere SDK to 20.07, and strangely it can't be seen in our environments.

Did you try this code with the previous Azure Sphere SDK 20.04?
And did you also try other sample code (like samples for freeRTOS) but the error still existed?

**Comment by kevinwh-chou** [comment]
Hi,

Can you try the following moves again with the attachment to see if it works or not? 
[cmake_20200901.zip](https://github.com/MediaTek-Labs/mt3620_m4_software/files/5155654/cmake_20200901.zip)

1. Replace the original CMakeLists.txt and CMakeSettings.json in the app folder with the attached ones. 
2. Copy the AzureSphereRTCoreToolchainMTK.cmake in the attachment to CMakeFiles folder in Azure Sphere SDK directory (like mine is C:\Program Files (x86)\Microsoft Azure Sphere SDK\CMakeFiles\). 

Thanks!

**Comment by matsujirushi** [comment]
Hi @kevinwh-chou ,
It work!

Copy cmake_20200901.zip to project folder.
(No need to copy AzureSphereRTCoreToolchainMTK.cmake.)

If you want, I can test all projects and create PR.

**Comment by LawranceLiu** [comment]
Hi Matsuoka-san, 

Thank you very much for your kindly support.
We have figured out some solutions (which are a little bit different for BareMetal and FreeRTOS) and will put to GitHub soon.

This issue seems to happen only on some Windows PC. In the past few days, I could not reproduce this build error problem until today I got a new PC and installed the Visual Studio & Azure Sphere SDK from scratch. This issue seems to be caused by some environment variable configuration changes due to SDK update. We will adjust the environment variable configurations in the CMake files.
